packages/contracts-bedrock/src/L1/OPContractsManager.sol

```diff
@@ -142,21 +136,21 @@ contract OPContractsManager is ISemver, Initializable {
     /// @notice Address of the ProtocolVersions contract shared by all chains.
     IProtocolVersions public immutable protocolVersions;
 
-    /// @notice The latest release of the OP Contracts Manager, as a string of the format `op-contracts/vX.Y.Z`.
-    string public latestRelease;
-
-    /// @notice Maps a release version to a contract name to it's implementation data.
-    mapping(string => mapping(string => Implementation)) public implementations;
+    // @notice L1 smart contracts release deployed by this version of OPCM. This is used in opcm to signal which version
+    // of the L1 smart contracts is deployed. It takes the format of `op-contracts/v*.*.*`.
+    string public l1ContractsReleaseVersion;
 
     /// @notice Maps an L2 Chain ID to the SystemConfig for that chain.
     mapping(uint256 => ISystemConfig) public systemConfigs;
 
     /// @notice Addresses of the Blueprint contracts.
     /// This is internal because if public the autogenerated getter method would return a tuple of
-    /// addresses, but we want it to return a struct. This is also set via `initialize` because
-    /// we can't make this an immutable variable as it is a non-value type.
+    /// addresses, but we want it to return a struct.
     Blueprints internal blueprint;
 
+    /// @notice Addresses of the latest implementation contracts.
+    ImplementationContracts public implementationContracts;
```

Comment:
Let's make visibility consistent here. I think we should do this and remove all the individual getters at the end, since this is easier to maintain

```suggestion
    Blueprints internal blueprint;

    /// @notice Addresses of the latest implementation contracts.
    Implementations internal implementation;
```

then the getters would be

```solidity
// this one we already have
function blueprints() public view returns (Blueprints memory) {
  return blueprint;
}

// this one is new
function implementations() public view returns (Implementations memory) {
  return implementation;
}
```

Reply:
Getters removed and `implementations()` getter added here: https://github.com/ethereum-optimism/optimism/pull/12851/commits/c8068686416d6eda365a44d2bd74c9b33ac736d7#diff-32c676c7a2cc25dc934a436546eccc2c3484b44491cc1569dce382ffb13b84c7R707